### PR TITLE
PROD-2311 - Give group and profile types unique label colors

### PR DESCRIPTION
**Pull request by jitendrabanjara1991**
### Jira Issue:
https://buddyboss.atlassian.net/browse/PROD-2311

### General Note
Keep all conversations related to this PR in the associated Jira issue(s). Do NOT add comment on this PR or edit this PR’s description.

### Notes to Developer
* Ensure the IDs (i.e. PROD-1) of all associated Jira issues are reference in this PR’s title
* Ensure that you have achieved the Definition of Done before submitting for review
* When this PR is ready for review, move the associate Jira issue(s) to “Needs Review” (or “Code Review” for Dev Tasks)

### Notes to Reviewer
* Ensure that the Definition of Done have been achieved before approving a PR
* When this PR is approved, move the associated Jira issue(s) to “Needs QA” (or “Approved” for Dev Tasks)

### CODE REVIEW

**Review by chetansatasiya — APPROVED** [review]

File: src/bp-core/admin/bp-core-admin-functions.php [hunk 1]
```diff
@@ -3367,3 +3370,47 @@ function bb_get_pro_fields_class() {
...
+ * Added new meta box as text and background color for member types label.
+ *
+ * @since BuddyBoss [BBVERSION]
+ *
+ * @param $post Post data object.
+ */
+function bb_member_type_labelcolor_metabox( $post ) {
+	$post_type         = isset( $post->post_type ) ? $post->post_type : 'bp-member-type';
+	$meta_data         = get_post_meta( $post->ID, '_bp_member_type_label_color', true );
+	$label_color_data  = ! empty( $meta_data ) ? maybe_unserialize( $meta_data ) : '';
```
Comment (KartikSuthar) [inline src/bp-core/admin/bp-core-admin-functions.php:3383]:
```suggestion
	$label_color_data  = ! empty( $meta_data ) ? maybe_unserialize( $meta_data ) : array();
```

File: src/bp-core/admin/js/settings-page.js [hunk 2]
```diff
@@ -2126,6 +2126,52 @@ window.bp = window.bp || {};
 					}
 				}
 			);
+			
+			// Run only post_type is member type and group type.
+			if ( 'undefined' !== typeof BP_ADMIN.post_type ) {
```
Comment (KartikSuthar) [inline src/bp-core/admin/js/settings-page.js:2131]: also check the `wpColorPicker()` is exists.

File: src/bp-core/classes/class-bp-admin-tab.php [hunk 3]
```diff
@@ -109,67 +109,84 @@ public function register_admin_script() {
...
-						'default_data'  => array(
-							'submit_button_message' => esc_js( __( 'Are you sure you want to import data? This action is going to alter your database. If this is a live website you may want to create a backup of your database first.', 'buddyboss' ) ),
-							'clear_button_message'  => esc_js( __( 'Are you sure you want to delete all Default Data content? Content that was created by you and others, and not by this default data installer, will not be deleted.', 'buddyboss' ) ),
-						),
-						'repair_forums' => array(
-							'validate_site_id_message' => esc_html__( 'Select site to repair the forums', 'buddyboss' ),
-						),
+			$localize_arg = array(
+				'ajax_url'            => admin_url( 'admin-ajax.php' ),
+				'select_document'     => esc_js( __( 'Please upload a file to check the MIME Type.', 'buddyboss' ) ),
```
Comment (KartikSuthar) [inline src/bp-core/classes/class-bp-admin-tab.php:114]:
```suggestion
				'select_document'     => esc_js( esc_html__( 'Please upload a file to check the MIME Type.', 'buddyboss' ) ),
```

File: src/bp-core/classes/class-bp-admin-tab.php [hunk 4]
```diff
@@ -109,67 +109,84 @@ public function register_admin_script() {
...
+				'tools'               => array(
+					'default_data'  => array(
+						'submit_button_message' => esc_js( __( 'Are you sure you want to import data? This action is going to alter your database. If this is a live website you may want to create a backup of your database first.', 'buddyboss' ) ),
```
Comment (KartikSuthar) [inline src/bp-core/classes/class-bp-admin-tab.php:117]:
```suggestion
						'submit_button_message' => esc_js( esc_html__( 'Are you sure you want to import data? This action is going to alter your database. If this is a live website you may want to create a backup of your database first.', 'buddyboss' ) ),
```

File: src/bp-core/classes/class-bp-admin-tab.php [hunk 5]
```diff
@@ -109,67 +109,84 @@ public function register_admin_script() {
...
+						'clear_button_message'  => esc_js( __( 'Are you sure you want to delete all Default Data content? Content that was created by you and others, and not by this default data installer, will not be deleted.', 'buddyboss' ) ),
```
Comment (KartikSuthar) [inline src/bp-core/classes/class-bp-admin-tab.php:118]:
```suggestion
						'clear_button_message'  => esc_js( esc_html__( 'Are you sure you want to delete all Default Data content? Content that was created by you and others, and not by this default data installer, will not be deleted.', 'buddyboss' ) ),
```

File: src/bp-core/classes/class-bp-admin-tab.php [hunk 6]
```diff
@@ -109,67 +109,84 @@ public function register_admin_script() {
...
-							'item_type'       => 'default',
-							'has_cover_image' => false,
-						),
-						'remove'            => array(
-							'nonce'  => wp_create_nonce( 'bp_delete_cover_image' ),
-							'action' => 'bp_cover_image_delete',
-							'json'   => true,
+				),
+				'moderation'          => array(
+					'suspend_confirm_message'   => esc_js( __( 'Please confirm you want to suspend this member. Members who are suspended will be logged out and not allowed to login again. Their content will be hidden from all members in your network. Please allow a few minutes for this process to complete.', 'buddyboss' ) ),
```
Comment (KartikSuthar) [inline src/bp-core/classes/class-bp-admin-tab.php:125]:
```suggestion
					'suspend_confirm_message'   => esc_js( esc_html__( 'Please confirm you want to suspend this member. Members who are suspended will be logged out and not allowed to login again. Their content will be hidden from all members in your network. Please allow a few minutes for this process to complete.', 'buddyboss' ) ),
```

File: src/bp-core/classes/class-bp-admin-tab.php [hunk 7]
```diff
@@ -109,67 +109,84 @@ public function register_admin_script() {
...
+					'unsuspend_confirm_message' => esc_js( __( 'Please confirm you want to unsuspend this member. Members who are unsuspended will be allowed to login again, and their content will no longer be hidden from other members in your network. Please allow a few minutes for this process to complete.', 'buddyboss' ) ),
```
Comment (KartikSuthar) [inline src/bp-core/classes/class-bp-admin-tab.php:126]:
```suggestion
					'unsuspend_confirm_message' => esc_js( esc_html__( 'Please confirm you want to unsuspend this member. Members who are unsuspended will be allowed to login again, and their content will no longer be hidden from other members in your network. Please allow a few minutes for this process to complete.', 'buddyboss' ) ),
```

File: src/bp-groups/bp-groups-admin.php [hunk 8]
```diff
@@ -2604,3 +2607,48 @@ function bp_group_type_set_platform_tab_submenu_active( $parent_file ) {
...
+ * Added new meta box as text and background color for group types label.
+ *
+ * @since BuddyBoss [BBVERSION]
+ *
+ * @param $post Post data object.
+ */
+function bb_group_type_labelcolor_metabox( $post ) {
+	$post_type         = isset( $post->post_type ) ? $post->post_type : 'bp-group-type';
+	$meta_data         = get_post_meta( $post->ID, '_bp_group_type_label_color', true );
+	$label_color_data  = ! empty( $meta_data ) ? maybe_unserialize( $meta_data ) : '';
```
Comment (KartikSuthar) [inline src/bp-groups/bp-groups-admin.php:2621]:
```suggestion
	$label_color_data  = ! empty( $meta_data ) ? maybe_unserialize( $meta_data ) : array();
```

**Review by KartikSuthar — CHANGES_REQUESTED** [review]

File: src/bp-core/classes/class-bp-admin-tab.php (hunk 3 above)
Comment (jitendrabanjara1991) [reply src/bp-core/classes/class-bp-admin-tab.php]: JI - This was not related to this PR but I have done this.

File: src/bp-core/classes/class-bp-admin-tab.php (hunk 4 above)
Comment (jitendrabanjara1991) [reply src/bp-core/classes/class-bp-admin-tab.php]: JI - This was not related to this PR but I have done this.

File: src/bp-core/classes/class-bp-admin-tab.php (hunk 5 above)
Comment (jitendrabanjara1991) [reply src/bp-core/classes/class-bp-admin-tab.php]: JI - This was not related to this PR but I have done this.

File: src/bp-core/classes/class-bp-admin-tab.php (hunk 6 above)
Comment (jitendrabanjara1991) [reply src/bp-core/classes/class-bp-admin-tab.php]: JI - This was not related to this PR but I have done this.

File: src/bp-core/classes/class-bp-admin-tab.php (hunk 7 above)
Comment (jitendrabanjara1991) [reply src/bp-core/classes/class-bp-admin-tab.php]: JI - This was not related to this PR but I have done this.

**Review by KartikSuthar — APPROVED** [review]
